tools/Normalise_Consent_Data.php

```diff
@@ -0,0 +1,247 @@
+                      WHERE ' . $consentName . ' IS NOT NULL OR ' . $consentName . ' != ""';
+        $psData    = $db->pselect($dateQuery, array());
+        foreach ($psData as $entry) {
+            if($entry[$consentDate] === "0000-00-00" || $entry[$consentWithdrawal] === "0000-00-00") {
+                array_push($errors, "Zero dates found in participant_status for:
+                           [ID]     => " . $entry['ID'] . "
+                           [CandID] => " . $entry['CandID'] . "
+                           Please remove date or run /tools/DB_date_zeros_removal.php.");
+            }
+        }
+    }
+}
+//////////////////
+// THROW ERRORS //
+//////////////////
+if (!empty($errors)) {
+    print_r($errors);
+    die("Resolve errors and run script again.\n");
+} else {
+    echo "\nValidation successful.\n";
+}
+
+/////////////////////
+// CONTINUE SCRIPT //
+/////////////////////
+
+// Update ConfigSetting table with value of 'useConsent' if true. Default is set to false.
+$configID = $db->pselectOne(
+    'SELECT ID FROM ConfigSettings WHERE Name="useConsent"',
+    array()
+);
+if ($useConsent === "true") {
+    $updateValue = ['Value' => $useConsent];
+    $db->update(
+        'Config',
+        $updateValue,
+        array (
+         'ConfigID' => $configID, 
+        )
+    );
+    echo "\nConfig settings set to use consent.\n\n";
+}
+
+/////////////////////////////////////////////
+// START IMPORT OF CONSENT INTO NEW TABLES //
+/////////////////////////////////////////////
+
+$consentType = [];
+$printArray  = [];
+foreach ($consents as $key=>$consent) {
+    $consentName       = $consent['name'];
+    $consentLabel      = $consent['label'];
+  
+    // Populate consent table with consents from Config.xml
+    $db->insert(
+        'consent', 
+        array(
+           'Name'  => $consentName,
+           'Label' => $consentLabel,
+         )
+    );
+    // Save ConsentID inserted
+    $consentID = $db->pselectOne(
+        'SELECT ConsentID FROM consent WHERE Name=:consentName',
+        array('consentName' => $consentName)
+    );
+    // Create array of consents to use later in importing history data
+    $consentList[$consentName] = $consentLabel;
+   
+    // Get all data where consent status has a value in participant_status
+    $consentDate       = $consentName . "_date";
+    $consentWithdrawal = $consentName . "_withdrawal";
+    $dataQuery         = 'SELECT CandID, ' . $consentName . ', ' . $consentDate . ', ' . $consentWithdrawal . ' FROM participant_status 
+                         WHERE ' . $consentName . ' IS NOT NULL OR ' . $consentName . ' != ""';
+    $psData            = $db->pselect($dataQuery, array());
+    print_r($psData);
+    foreach ($psData as $entry) {
+        // Import 'not_answered' status as NULL
+        if ($entry[$consentName] === "not_answered") {
+            $entry[$consentName] = "";
+        }
```

Comment:
wouldn't it make more sense to just not import "not_answered"? The discussion at the last LORIS meeting seemed to agree on not answered not making any sense for consent.

Reply:
but what if a candidate has "not_answered" and you want to import the consent date given and date withdrawn attached to it? are we okay with removing all data involved?